apollo-ios-codegen/Sources/ApolloCodegenLib/Capitalizer.swift

```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+public enum CapitalizationRule: Codable, Equatable {
+  case uppercase(regex: String)
```

Comment:
Wondering if a regex pattern is the right way to handle this vs allowing explicit declaration of the type you want to change capitalization for and what you want it to be through a dictionary similar to how we handle the schema type renaming.

Thoughts @AnthonyMDev @calvincestari ?

Reply:
I'm not sure. That's a good question. But we aren't just talking about types here, we're also talking about their properties. 

If you used `.uppercase(regex: "id"), that could also result in situations where a property `isHidden` gets changed to `isHIDden`, for example. You would need to make your regex `^id$` if you wanted to just match the exact string `id`, which I guess is alright, but not super easy to figure out if you're not very familiar with regex.

However, I think it's likely that there will be situations where you want to tell the codegen engine "uppercase all instances of the letters 'UUID'. This may show up in multiple different property names, and is less likely to have conflicts where it shows up in the middle of another word. A regex would be helpful in those situations.

There is also situations where you want to capitalize a term in a specific way, like `GraphQL`. I think there should be a way to do that too. This doesn't fall into `camelcase` exactly correctly.

---

I think a more robust API would allow users to use _either a regex or a direct string literal_ and allow them to choose a common casing strategy or a direct replacement string.

```swift
struct CapitalizationRule {

  enum Term {
    case string(String)
    case regex(String)
  }

  enum Strategy {
    case uppercase
    case lowercase
    case titlecase
    case camelcase
    case pascalcase
    case replace(String)
  }

  let term: Term
  let strategy: Strategy
}
```

This would allow for all of the desired capabilities to exist. My only concern is that the `replace` strategy allows for way more use cases outside of just capitalization. It might be used as a hacky workaround for other desired name changes that are not the intended usage of this API.

What do you think @tonyarnold @BobaFetters?

Reply:
I quite like this proposed API! I'll update the PR to reflect it. 

We could potentially leave `.replace(…)` out of the strategies?